core/src/main/java/org/apache/carbondata/scan/filter/FilterExpressionProcessor.java

```diff
@@ -286,6 +289,13 @@ private FilterResolverIntf getFilterResolverBasedOnExpressionType(
               return new RowLevelFilterResolverImpl(expression, isExpressionResolve, true,
                   tableIdentifier);
             }
+            if (currentCondExpression.getFilterExpressionType() == ExpressionType.CONTAINS
```

Comment:
Row evaluation will happen in case of no dictionary scenario, and this scenario is applicable for dictionary columns.